apkid/rules/dex/protectors.yara

```diff
@@ -111,3 +111,32 @@ rule appdome_dex : protector
   condition:
     is_dex and $loader
 }
+
+rule free_rasp_dex : protector
+{
+  meta:
+    description = "FreeRASP"
+    sample      = "e10b8772fd9b6aaf8ba030c5bcb324fb9b91f34e893a62bdf238629df856e047"
+    url         = "https://www.talsec.app/freerasp-in-app-protection-security-talsec"
+    author      = "Fare9"
+  
+  strings:
+    // Decryption method found in DEX files, since strings will change
+    // and other offsets change, we add ?? to some instructions
+    $decryption = {
+      6e 10 ?? ?? 08 00 0a 00 db 00 00 02 23 01 ?? ??
```

Comment:
Would it be possible to display commented offsets this way?  This is only to make it clear to read in the future by others.
```sh
    // Most of some kind of runtime decryption method, signature = a(IIZI[I[[I[I)V
    $decrypt_method = {
      12 01                 // const/4 v1, 0x0
      39 ?? ?? ??           // if-nez ??, :????
      71 ?? ?? ?? ?? ??     // invoke-static {??}, ??
      01 10                 // move v0, v1
      35 ?? ?? ??           // if-ge ?, ?, :????
      44 ?? ?? ??           // aget ??, ??, ??
      B7 ??                 // xor-int/2addr (xor is fairly rare in legit code)
      71 ?? ?? ?? ?? ??     // invoke-static
      0A ??                 // move-result ??
      97 ?? ?? ??           // xor-int ??, ??, ??
      D8 00 00 01           // add-int/lit8 v0, v0, 0x1
      01 ??                 // move ?, ?
      28 F2                 // goto
      21 80                 // array-length
      D8 00 00 FE           // add-int/lit8 v0, v0, -0x2
      44 ?? ?? ??           // aget ??
      B7 ??                 // xor-int/2addr
      21 ??                 // invoke-static
      D8 ?? ?? ??           // add-int
      44 ?? ?? ??           // aget
      B7 ??                 // xor-int/2addr
    }
```